service/lib/agama/proxy_setup.rb

```diff
@@ -30,23 +32,39 @@ class ProxySetup
     include Singleton
     include Yast
     include Logger
+    include Helpers
 
     CMDLINE_PATH = "/proc/cmdline"
     CMDLINE_MENU_CONF = "/etc/cmdline-menu.conf"
+    PACKAGES = ["microos-tools"].freeze
+    CONFIG_PATH = "/etc/sysconfig/proxy"
+    PROPOSAL_ID = "network_proposal"
 
     # @return [URI::Generic]
     attr_accessor :proxy
 
     # Constructor
     def initialize
       Yast.import "Proxy"
+      Yast.import "Installation"
+      Yast.import "PackagesProposal"
     end
 
     def run
       read
       write
     end
 
+    def install
+      on_local do
+        Proxy.Read
+        return unless Proxy.enabled
+
+        copy_files
+        add_packages
```

Comment:
We probably will need to do it separately as the packages should be added before the software is installed but then the microos-tools https://github.com/openSUSE/microos-tools/blob/master/systemd-proxy-env/setup-systemd-proxy-env.path need to be enabled later as well as the copy of the files.